src/main/java/kr/codesqaud/cafe/account/UserService.java

```diff
@@ -22,44 +24,68 @@ public UserService(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
 
-    public int save(JoinForm joinForm) {
+    public User save(JoinForm joinForm) {
         User user = joinForm.toUser();
         return userRepository.save(user);
     }
 
     public List<UserForm> getAllUsersForm() {
-        return userRepository.getAllUsers().stream()
+        return userRepository.findAll().stream()
                 .map(UserForm::from)
                 .collect(Collectors.toList());
     }
 
-    public void update(ProfileSettingForm profileSettingForm, Long userId) {
-        userRepository.update(profileSettingForm.setUser(findById(userId)));
+    public User update(User user, ProfileEditForm profileEditForm) {
+        User entity = profileEditForm.setUser(user);
+        return userRepository.save(entity);
     }
 
-    public boolean isSamePassword(Long userId, String targetPassword) {
-        Optional<User> userOptional = userRepository.findById(userId);
-        return userOptional.map(user -> user.isSamePassword(targetPassword)).orElse(false);
+    public boolean isSamePassword(User user, String targetPassword) {
+        return user.isSamePassword(targetPassword);
     }
 
     public Optional<User> findByEmail(String email) {
         return userRepository.findByEmail(email);
     }
 
     public boolean containsEmail(String email) {
-        return userRepository.containsEmail(email);
+        return userRepository.existsByEmail(email);
+    }
+
+    public Optional<User> findById(Long userId) {
+        return userRepository.findById(userId);
     }
 
-    public User findById(Long userId) {
-        Optional<User> userOptional = userRepository.findById(userId);
-        if (userOptional.isPresent()) {
-            return userOptional.get();
-        } else {
-            throw new InvalidUserIdException(INVALID_USER_ID_CODE);
+    public void checkId(User user, Long userId) {
+        if (!user.isSameId(userId)) {
+            throw new RuntimeException("접근 할 수 없습니다.");
         }
     }
 
-    public boolean isDuplicateEmail(String targetEmail) {
-        return userRepository.findByEmail(targetEmail).isPresent();
+    public void checkManager(User user) {
+        if (!user.isManager()) {
+            throw new RuntimeException("접근 할 수 없습니다.");
+        }
+    }
```

Comment:
```suggestion
    // User.class
    public void checkManager(User user) {
    if (!role.equals(Role.MANAGER)) {
            throw new RuntimeException("접근 할 수 없습니다.");
        }
    }
```
Repository, DB 접근을 하지않는다면 도메인객체에게 묻는것을 어떤가요? 이렇게 했을 경우 컨트롤러에서도 `userSerivce.checkManager` 가 아닌 `user.checkManager`로 객체에게 확인을 받을 수 있습니다. 이렇게 했을 때 장점은 서비스에서 관리하지 않고 도메인에서 관리할 수 있고 테스트할 때 도메인 단위테스트만 하면 되기 때문에 테스트가 편리합니다.

조금 이해하기 어려우실 수 있는데 서비스에서는 실제 비즈니스코드를 쓰기보다는 도메인간 의사소통을 교통정리한다고 생각하시면 됩니다.

여기 관련해서는 이동욱님 블로그 및 서적(스프링 부트와 AWS로 혼자 구현하는 웹 서비스)를 검색하시는걸 추천합니다.  지금 이해하기 어려우시다면 이런게 있구나만 확인하시고 넘기시면 됩니다.

Reply:
- 감사합니다. 찾아보고 확인하겠습니다.

Reply:
- 조금 이해하기 힘든 부분인 것 같습니다. 책을 구매해서 한번 읽어 보겠습니다. 감사합니다.🙇🏻‍♂️